### PR TITLE
Add eslint to loopback-workspace

### PR DESCRIPTION
**Pull request by 0candy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
   },
   "scripts": {
     "pretest": "test -z \"$JENKINS_HOME\" || (echo | node bin/setup-mysql.js)",
-    "test": "mocha"
+    "test": "mocha",
+    "posttest": "eslint"
```
Comment (superkhau) [inline package.json]:
```
"lint": "eslint",
"test": "mocha",
"posttest": "npm run lint"
```

**Comment by superkhau** [comment]
One thing to fix, otherwise LGTM.

File: common/models/config-file.js [hunk 2]
```diff
@@ -197,49 +196,48 @@ module.exports = function(ConfigFile) {
...
     var dir = this.getDirName();
     // NOTE: glob always returns the path using forward-slash even on Windows
     // See: https://github.com/strongloop/generator-loopback/issues/12
     var baseDir = this.path.split('/')[0];
 
-    if(dir === ROOT_COMPONENT
-      || baseDir === this.path
-      || baseDir === 'models') {
+    if (dir === ROOT_COMPONENT || baseDir === this.path ||
+      baseDir === 'models') {
```
Comment (bajtos) [inline common/models/config-file.js]: See https://github.com/strongloop-internal/loopback-knowledge-base/blob/4772f51fcc68c7c1a83157b31eac5379e509d238/style-guide.md#indentation-of-multi-line-expressions-in-return

```js
if (dir === ROOT_COMPONENT
    || baseDir === this.path
    || baseDir === 'models') {
```

even better:

```js
var isRootComponent = dir === ROOT_COMPONENT
  || baseDir === this.path
  || baseDir === 'models';

return isRootComponent ? ROOT_COMPONENT : baseDir;
```

File: common/models/config-file.js [hunk 3]
```diff
@@ -248,10 +246,11 @@ module.exports = function(ConfigFile) {
 
       cb(null, result);
     });
-  }
+  };
 
   function entityBelongsToFacet(name, definition) {
-    return definition && definition.properties && definition.properties.facetName;
+    return definition && definition.properties &&
+      definition.properties.facetName;
```
Comment (bajtos) [inline common/models/config-file.js]: Please split into three lines, see https://github.com/strongloop-internal/loopback-knowledge-base/blob/4772f51fcc68c7c1a83157b31eac5379e509d238/style-guide.md#indentation-of-multi-line-expressions-in-return (the code examples don't explicitly say it, would you mind submitting a PR to fix that?)

```js
return definition && 
  definition.properties &&
  definition.properties.facetName;
```

Or perhaps:

```js
var properties = definition && definition.properties;
return properties && properties.facetName;
```

Comment (bajtos) [inline common/models/config-file.js]: > Please split into three lines, see https://github.com/strongloop-internal/loopback-knowledge-base/blob/4772f51fcc68c7c1a83157b31eac5379e509d238/style-guide.md#indentation-of-multi-line-expressions-in-return (the code examples don't explicitly say it, would you mind submitting a PR to fix that?)

Hmm, on second thought, that may not be the best idea. Let's either keep what you have now (and ignore my comment entirely), or refactor it as shown in the second snippet.

```js
var properties = definition && definition.properties;
return properties && properties.facetName;
```

File: common/models/config-file.js [hunk 4]
```diff
@@ -305,17 +303,16 @@ module.exports = function(ConfigFile) {
...
     var configFile;
     var results = [];
-    for(var i = 0; i < configFiles.length; i++) {
+    for (var i = 0; i < configFiles.length; i++) {
       configFile = configFiles[i];
       // TODO(ritch) support other directories
-      if(configFile && configFile.getFacetName() === facetName
-        && configFile.getDirName() === 'models') {
+      if (configFile && configFile.getFacetName() === facetName &&
+        configFile.getDirName() === 'models') {
```
Comment (bajtos) [inline common/models/config-file.js]: https://github.com/strongloop-internal/loopback-knowledge-base/blob/4772f51fcc68c7c1a83157b31eac5379e509d238/style-guide.md#indentation-of-multi-line-expressions-in-if

Indent the second line by two levels.

File: common/models/middleware.js [hunk 5]
```diff
@@ -93,13 +92,13 @@ function ready(Middleware) {
...
       }
       var phasePlaceHolder =
-      {
-        phase: phase,
-        isPhasePlaceHolder: true,
-        order: order,
-        facetName: facetName,
-        name: '*' + phase // Set the name to be unique
-      };
+        {
```
Comment (bajtos) [inline common/models/middleware.js]: I personally prefer this style:

```js
var phasePlaceHolder = {
  phase: phase,
  // ...
};
```

It should also minimise the amount of changes needed to make eslint happy.

File: common/models/config-file.js (hunk 3 above)
Comment (superkhau) [inline common/models/config-file.js]:
```
var facetName = definition
  && definition.properties
  && definition.properties.facetName;

return facetName;
```

File: common/models/config-file.js (hunk 2 above)
Comment (superkhau) [inline common/models/config-file.js]:
```
var isRootComponent = dir === ROOT_COMPONENT
  || baseDir === this.path
  || baseDir === 'models';

var facetName = isRootComponent ? ROOT_COMPONENT : baseDir;

return facetName;
```

I prefer not having to parse the details to figure out what the intent of the function is (ie. you are trying to return a facet name, i don't need to see the guts `return isRootComponent ? ...` to determine this.

File: common/models/config-file.js (hunk 3 above)
Comment (bajtos) [inline common/models/config-file.js]: @superkhau ^^^ that does not work, `&&` must be at the end of the line, not at the beginning as in your example.

**Comment by bajtos** [comment]
LGTM, 🚢 

@0candy in the future, when modifying a patch as large as this one, it's better to add small `fixup` commits rather than amending the single huge one. That way the reviewer can quickly see what has been changed since the last review, compared to the current situation, when I had to scan the full patch again.

**Comment by superkhau** [comment]
> in the future, when modifying a patch as large as this one, it's better to add small fixup commits rather than amending the single huge one. That way the reviewer can quickly see what has been changed since the last review, compared to the current situation, when I had to scan the full patch again.

This should be KB.

File: common/models/config-file.js (hunk 3 above)
Comment (superkhau) [inline common/models/config-file.js]: Brainfart, you are correct.

```
var facetName = definition &&
  definition.properties &&
  definition.properties.facetName;

return facetName;
```

**Pull request merged**